ec/ecresource/deploymentresource/elasticsearch/v2/elasticsearch.go

```diff
@@ -0,0 +1,413 @@
+	TrustExternal    types.Set    `tfsdk:"trust_external"`
+	Strategy         types.String `tfsdk:"strategy"`
+}
+
+type Elasticsearch struct {
+	Autoscale        *string                      `tfsdk:"autoscale"`
+	RefId            *string                      `tfsdk:"ref_id"`
+	ResourceId       *string                      `tfsdk:"resource_id"`
+	Region           *string                      `tfsdk:"region"`
+	CloudID          *string                      `tfsdk:"cloud_id"`
+	HttpEndpoint     *string                      `tfsdk:"http_endpoint"`
+	HttpsEndpoint    *string                      `tfsdk:"https_endpoint"`
+	HotTier          *ElasticsearchTopology       `tfsdk:"hot"`
+	CoordinatingTier *ElasticsearchTopology       `tfsdk:"coordinating"`
+	MasterTier       *ElasticsearchTopology       `tfsdk:"master"`
+	WarmTier         *ElasticsearchTopology       `tfsdk:"warm"`
+	ColdTier         *ElasticsearchTopology       `tfsdk:"cold"`
+	FrozenTier       *ElasticsearchTopology       `tfsdk:"frozen"`
+	MlTier           *ElasticsearchTopology       `tfsdk:"ml"`
+	Config           *ElasticsearchConfig         `tfsdk:"config"`
+	RemoteCluster    ElasticsearchRemoteClusters  `tfsdk:"remote_cluster"`
+	SnapshotSource   *ElasticsearchSnapshotSource `tfsdk:"snapshot_source"`
+	Extension        ElasticsearchExtensions      `tfsdk:"extension"`
+	TrustAccount     ElasticsearchTrustAccounts   `tfsdk:"trust_account"`
+	TrustExternal    ElasticsearchTrustExternals  `tfsdk:"trust_external"`
+	Strategy         *string                      `tfsdk:"strategy"`
+}
+
+func ElasticsearchPayload(ctx context.Context, esObj types.Object, template *models.DeploymentTemplateInfoV2, dtID, version string, useNodeRoles bool, skipTopologies bool) (*models.ElasticsearchPayload, diag.Diagnostics) {
+	var es *ElasticsearchTF
+
+	if esObj.IsNull() || esObj.IsUnknown() {
+		return nil, nil
+	}
+
+	if diags := tfsdk.ValueAs(ctx, esObj, &es); diags.HasError() {
+		return nil, diags
+	}
+
+	if es == nil {
+		return nil, nil
+	}
+
+	if es == nil {
+		var diags diag.Diagnostics
+		diags.AddError("Elasticsearch payload error", "cannot find elasticsearch data")
+		return nil, diags
+	}
+
+	templatePayload := utils.EnrichElasticsearchTemplate(utils.EsResource(template), dtID, version, useNodeRoles)
+
+	payload, diags := es.Payload(ctx, templatePayload, skipTopologies)
+	if diags.HasError() {
+		return nil, diags
+	}
+
+	return payload, nil
+}
+
+func ReadElasticsearches(in []*models.ElasticsearchResourceInfo, remotes *models.RemoteResources) (*Elasticsearch, error) {
+	for _, model := range in {
+		if util.IsCurrentEsPlanEmpty(model) || utils.IsEsResourceStopped(model) {
+			continue
+		}
+		es, err := ReadElasticsearch(model, remotes)
+		if err != nil {
+			return nil, err
+		}
+		return es, nil
+	}
+
+	return nil, nil
+}
+
+func ReadElasticsearch(in *models.ElasticsearchResourceInfo, remotes *models.RemoteResources) (*Elasticsearch, error) {
```

Comment:
This doesn't need to be exported. 

```suggestion
func readElasticsearch(in *models.ElasticsearchResourceInfo, remotes *models.RemoteResources) (*Elasticsearch, error) {
```